src/tests/ngraph_helpers/snippets_ngraph_functions/include/snippets_helpers.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+
+#include "openvino/core/model.hpp"
+#include "common_test_utils/ngraph_test_utils.hpp"
+
+namespace ngraph {
+namespace builder {
+namespace subgraph {
+using ov::Model;
+
+/// \brief Base class for snippets-related subgraphs. Note that inputShapes size is model-specific
+/// and expected to be checked inside a child class constructor.
+/// \param inputShapes vector of input shapes accepted by the underlying model
+class SnippetsFunctionBase {
```

Comment:
Why do we need base type for function builders? I think if we remove `SnippetsFunctionBase` then we will have more simpler source code.
For example: does base test type have pointer to the function builder instance to run virtual methods? - I didn't find it in existing source code.

Reply:
As we discussed earlier https://github.com/openvinotoolkit/openvino/pull/10605#discussion_r827767764, let's keep it for now.
I also provided [an example](https://github.com/IvanNovoselov/openvino/blob/aa40b2b7e5976560bfd7ae699957b69bca7ef30f/src/tests/functional/inference_engine/snippets/src/pass/insert_load_store.cpp#L36-L40) where a pointer to the base class is used, although I agree that it is not absolutely essential in this case, it still makes code more concise and readable.

Reply:
I'm not sure we should do it because the lowering will be performed in SetUp in this case. We however need to test the lowering, so it should be performed inside TEST_P instead. Please see also this reply https://github.com/openvinotoolkit/openvino/pull/10605#discussion_r830815416